test/common/common.go

```diff
@@ -0,0 +1,230 @@
+				"loglevel.autoscaler": "debug",
+			},
+		},
+	}
+	// Update it
+	if ks, err = clients.KnativeServing().Update(ks); err != nil {
+		t.Fatalf("KnativeServing %q failed to update: %v", names.KnativeServing, err)
+	}
+	// Verify the relevant configmaps have been updated
+	err = resources.WaitForConfigMap(defaultsConfigMapName, clients.KubeClient.Kube, func(m map[string]string) bool {
+		return m["revision-timeout-seconds"] == "200"
+	})
+	if err != nil {
+		t.Fatalf("The operator failed to update %s configmap", defaultsConfigMapName)
+	}
+	err = resources.WaitForConfigMap(loggingConfigMapName, clients.KubeClient.Kube, func(m map[string]string) bool {
+		return m["loglevel.controller"] == "debug" && m["loglevel.autoscaler"] == "debug"
+	})
+	if err != nil {
+		t.Fatalf("The operator failed to update %s configmap", loggingConfigMapName)
+	}
+
+	// Delete a single key/value pair
+	delete(ks.Spec.Config[loggingConfigKey], "loglevel.autoscaler")
+	// Update it
+	if ks, err = clients.KnativeServing().Update(ks); err != nil {
+		t.Fatalf("KnativeServing %q failed to update: %v", names.KnativeServing, err)
+	}
+	// Verify the relevant configmap has been updated
+	err = resources.WaitForConfigMap(loggingConfigMapName, clients.KubeClient.Kube, func(m map[string]string) bool {
+		_, autoscalerKeyExists := m["loglevel.autoscaler"]
+		// deleted key/value pair should be removed from the target config map
+		return m["loglevel.controller"] == "debug" && !autoscalerKeyExists
+	})
+	if err != nil {
+		t.Fatal("The operator failed to update the configmap")
+	}
+
+	// Use an empty map as the value
+	ks.Spec.Config[defaultsConfigKey] = map[string]string{}
+	// Update it
+	if ks, err = clients.KnativeServing().Update(ks); err != nil {
+		t.Fatalf("KnativeServing %q failed to update: %v", names.KnativeServing, err)
+	}
+	// Verify the relevant configmap has been updated and does not contain any keys except "_example"
+	err = resources.WaitForConfigMap(defaultsConfigMapName, clients.KubeClient.Kube, func(m map[string]string) bool {
+		_, exampleExists := m["_example"]
+		return len(m) == 1 && exampleExists
+	})
+	if err != nil {
+		t.Fatal("The operator failed to update the configmap")
+	}
+
+	// Now remove the config from the spec and update
+	ks.Spec = v1alpha1.KnativeServingSpec{}
+	if ks, err = clients.KnativeServing().Update(ks); err != nil {
+		t.Fatalf("KnativeServing %q failed to update: %v", names.KnativeServing, err)
+	}
+	// And verify the configmap entry is gone
+	err = resources.WaitForConfigMap(loggingConfigMapName, clients.KubeClient.Kube, func(m map[string]string) bool {
+		_, exists := m["loglevel.controller"]
+		return !exists
+	})
+	if err != nil {
+		t.Fatal("The operator failed to revert the configmap")
+	}
+}
+
+// DeploymentRecreation verify whether all the deployments for knative serving are able to recreate, when they are deleted.
+func DeploymentRecreation(t *testing.T, clients *test.Clients, names test.ResourceNames) {
```

Comment:
would break this into separate methods
deleteDeployments
verifyDeployments.

Reply:
you can keep this method, but rename it DeleteAndVerifyDeployments, to make it clear all that the method is doing.